### PR TITLE
Adjust on list price update with cny_rate

### PR DESCRIPTION
**Pull request by eHanseFrank**
- adding a new calculated field to product.template: net_price_rmb which uses cny_rate in its compute method [NOT WORKING]
- further adjustments to the attribute order in view

### CODE REVIEW

File: product_listprice_list_view/models/product_template.py [hunk 1]
```diff
@@ -133,10 +139,15 @@ def _compute_net_profit(self):
...
-
     @api.multi
     def _get_cny_rate(self):
         cny_rec = self.env['res.currency'].search([('name', '=', 'CNY')])[0]
         if cny_rec:
             for pt in self:
                 pt.cny_rate = cny_rec.rate_silent
+
+    @api.multi
+    @api.depends('cny_rate','net_price')
```
Comment (yostashiro) [inline product_listprice_list_view/models/product_template.py:150]: No need to have `@api.depends` for computation for non-stored field

File: product_listprice_list_view/models/product_template.py [hunk 2]
```diff
@@ -133,10 +139,15 @@ def _compute_net_profit(self):
...
+    def _get_net_price_rmb(self):
+        for pt in self:
+            pt.net_price_rnb = pt.net_price * pt.cny_rate
```
Comment (yostashiro) [inline product_listprice_list_view/models/product_template.py:153]: Typo here - should be `pt.net_price_rmb` instead of `pt.net_price_rnb`

Comment (yostashiro) [reply product_listprice_list_view/models/product_template.py:153]: I would actually include this calculation in `_get_cny_rate` method

File: product_listprice_list_view/views/product_template_views.xml [hunk 3]
```diff
@@ -9,20 +9,20 @@
...
                     <field name="advertise" string="Ads"/>
                     <field name="stock_location"/>
                     <field name="stock_leadtime"/>
-                    <field name="last_in_date" string="Incoming Date" widget="date"/>
+                    <field name="stock_cost" string="Cost HKD"/>
+                    <field name="net_profit" string="Net Profit" fg_color="red:net_profit &lt; 0.00;"/>
+                    <field name="net_profit_pct" string="Net Profit %" />
                     <field name="qty_reserved" string="Qty reserved"/>
-                    <field name="qty_local_atp" string="Qty internal"/>
+                    <field name="qty_available" string="Qty internal"/>
```
Comment (yostashiro) [inline product_listprice_list_view/views/product_template_views.xml:23]: Add `widget="integer"`?

**Review by yostashiro — CHANGES_REQUESTED** [review]
@eHanseFrank I've added comments.  Please review and update your code. :D

**Comment by yostashiro** [comment]
@eHanseFrank Are you still following up on this topic?

**Comment by eHanseFrank** [comment]
You can close this. Thanks


Am 30-03-2017 um 7:05 PM schrieb Yoshi Tashiro:

**Pull request closed without merging**